### PR TITLE
[Candidate, TimePoint] Refactor setData() to accept only array as argument

### PR DESCRIPTION
**Pull request by johnsaigle**
### Brief summary of changes

Previously the setData() function in both of these classes accepted arrays or two separate strings. This prevents being able to use return types in these classes.

These functions were refactored to accept key-value arrays as they are occasionally called with multiple values to update. Calling functions are changed to use this new method of passing arguments.

### Related

* #4049 
* #4047 

### To test this change...

- [ ] Modify data in for Candidates and TimePoints. Everything should work as before.

### CODE REVIEW

File: php/libraries/TimePoint.class.inc [hunk 1]
```diff
@@ -330,29 +330,39 @@ class TimePoint
...
+     * @param array $newData Key-value pair(s) of the field(s) to change and its
+     *                              new value(s).
      *
-     * @return void (side-effect modifies session table.)
+     * @return bool true
+     *
+     * @throws DatabaseException
      */
-    function setData($var, $value = null)
+    public function setData(array $newData): bool
```
Comment (driusan) [inline php/libraries/TimePoint.class.inc]: Is this called from anywhere that it can't be made protected?

**Comment by johnsaigle** [comment]
@driusan I'm not sure but I'll change it to `protected` and see if phan etc. has any issues.

File: php/libraries/TimePoint.class.inc (hunk 1 above)
Comment (johnsaigle) [reply php/libraries/TimePoint.class.inc]: There are 3 instances in `next_stage` that use this function in a static context and are incompatible with `protected` according to phan.

File: modules/new_profile/php/new_profile.class.inc [hunk 2]
```diff
@@ -90,7 +90,12 @@ class New_Profile extends \NDB_Form
         $candidate =& \Candidate::singleton($candID);
 
         if ($config->getSetting('useProjects') == "true") {
-            $candidate->setData('ProjectID', $values['ProjectID']);
+            $candidate->setData(
+                array(
+                 'ProjectID',
+                 $values['ProjectID'],
+                )
```
Comment (driusan) [inline modules/new_profile/php/new_profile.class.inc]: Why is this one not an associative array?

**Comment by driusan** [comment]
@johnsaigle this has conflicts now

File: modules/new_profile/php/new_profile.class.inc (hunk 2 above)
Comment (johnsaigle) [reply modules/new_profile/php/new_profile.class.inc]: You're right, it should be.

**Comment by johnsaigle** [comment]
@driusan I've addressed your feedback and rebased. Please re-review when possible.

**Comment by driusan** [comment]
@johnsaigle needs to be rebased now because of conflicts.

**Pull request merged**